src/utils/chartOptions.js

```diff
@@ -0,0 +1,284 @@
+      },
+    }],
+  },
+
+  elements: {
+    rectangle: {
+      backgroundColor: chartStyles.ultramarineBlue,
+      borderWidth: '0',
+      borderColor: '',
+      borderSkipped: 'bottom',
+    },
+  },
+}, baseOptions, options);
+
+
+/**
+ * Options ONLY for Doughnut chart
+ * @param {object} options - More options that can be pass to the chart
+ */
+export const doughnutChartOptions = options => merge({
+  cutoutPercentage: 60,
+
+  elements: {
+    arc: {
+      backgroundColor: chartStyles.ultramarineBlue,
+      borderAlign: 'center',
+      borderColor: chartStyles.whiteColor,
+      borderWidth: 1,
+    },
+  },
+}, baseOptions, options);
+
+
+/**
+ * data ONLY for Line chart
+ * @param {object} data - More data that can be pass to the chart
+ */
+export const lineChartData = data => merge({
+  datasets: (data.datasets || []).map((_, index) => ({
+    backgroundColor: chartStyles.transparent,
+    borderColor: colorPallete[index],
+    pointBorderColor: colorPallete[index],
+    pointBackgroundColor: chartStyles.whiteColor,
+    pointHoverBackgroundColor: chartStyles.whiteColor,
+    pointHoverBorderColor: colorPallete[index],
+    pointHoverBorderWidth: 4,
+    borderWidth: 2,
+  })),
+}, data);
+
+
+/**
+ * data ONLY for Bar chart
+ * @param {object} data - More data that can be pass to the chart
+ */
+export const barChartData = data => merge({
+  datasets: (data.datasets || []).map((_, index) => ({
+    barPercentage: 1,
+    categoryPercentage: 0.5,
+    barThickness: 'flex',
+    maxBarThickness: 14,
+    minBarLength: 1,
+    stack: index,
+    backgroundColor: colorPallete[index],
+  })),
+}, data);
+
+
+/**
+ * data ONLY for Doughnut chart
+ * @param {object} data - More data that can be pass to the chart
+ */
+export const doughnutChartData = data => merge({
+  datasets: [
+    {
+      backgroundColor: colorPallete,
+    },
+  ],
+}, data);
+
+
+/**
+ * Function that return the corresponding options object
+ * based on the selected chart type.
+ * @param {string} type - can be line, bar or doughnut
+ * @param {object} options - More options that can be pass to the chart
+ */
+export const optionsByChart = (type, options) => {
+  switch (type) {
+    case typeLine: {
+      return lineChartOptions(options);
+    }
+
+    case typeBar: {
+      return barChartOptions(options);
+    }
+
+    case typeDoughnut: {
+      return doughnutChartOptions(options);
+    }
+
+    default:
+      return [];
+  }
+};
+
+
+/**
+ * Function that return the corresponding data object
+ * based on the selected chart type.
+ * @param {string} type - can be line, bar or doughnut
+ * @param {object} data - More data that can be pass to the chart
+ */
+export const dataByChart = (type, data) => {
+  switch (type) {
+    case typeLine: {
+      return lineChartData(data);
+    }
+
+    case typeBar: {
+      return barChartData(data);
+    }
+
+    case typeDoughnut: {
+      return doughnutChartData(data);
+    }
+
+    default:
+      return [];
```

Comment:
Not that it has some effect now, but for the future 🙂:
```suggestion
      return data;
```